### ISSUE TITLE
Display inter-function dependencies

### ISSUE THREAD

**Issue by odoodo** [odoodo/Mirth-Migrator#4]
**Context**
Mirth Migrator automatically detects and shows all directly or indirectly used functions of a channel:
![image](https://github.com/user-attachments/assets/b215aac1-1a6b-424c-ab60-80a197e93334)
This is however currently not the case for functions or code templates that use other functions.

**Goal**
Add support for automated detection an display of directly or indirectly used functions and code templates by other code templates and functions

**Needed Action**

- Basically just use the same routine (function _getFunctionUsesFunctions()_) that has been implemented for the function dependency detection of the channels. I might have to be slightly adapted to fit for both cases.
- Further, analog to the channel migration, an option for also migrating all referenced functions with the respective code template/function has to be added:
![image](https://github.com/user-attachments/assets/b86f5ba5-7b3c-4833-8d0e-5e10c0460d37)
